iet.so/include/iet_worker.h

```diff
@@ -62,7 +62,12 @@ class IETWorker : public rvs::ThreadBase {
     }
     //! returns the GPU index
     int get_gpu_device_index(void) { return gpu_device_index; }
-
+    //! sets the GPU smi index
+    void set_smi_device_index(int _smi_device_index) {
```

Comment:
void set_smi_device_index(int _smi_device_index)   should always be

void set_smi_device_index(const int&  _smi_device_index)

Reply:
for simpler and native data types like int, char , I dont think pass by const ref will differ much from pass by value

Reply:
agreed,  but like 'winning is a habit' so is 'good coding ways' should also be :)
you may leave the code currently as it is